src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -7,37 +7,35 @@
 import kitchenpos.menu.domain.MenuGroupRepository;
 import kitchenpos.menu.domain.MenuProduct;
 import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.domain.MenuValidator;
 import kitchenpos.menu.dto.MenuRequest;
 import kitchenpos.menu.dto.MenuResponse;
-import kitchenpos.product.domain.Product;
-import kitchenpos.product.domain.ProductRepository;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class MenuService {
     private final MenuRepository menuRepository;
     private final MenuGroupRepository menuGroupRepository;
-    private final ProductRepository productRepository;
+    private final MenuValidator menuValidator;
 
     public MenuService(
             final MenuRepository menuRepository,
             final MenuGroupRepository menuGroupRepository,
-            final ProductRepository productRepository
+            final MenuValidator menuValidator
     ) {
         this.menuRepository = menuRepository;
         this.menuGroupRepository = menuGroupRepository;
-        this.productRepository = productRepository;
+        this.menuValidator = menuValidator;
     }
 
     @Transactional
     public MenuResponse create(final MenuRequest menuRequest) {
-
-        final MenuGroup menuGroup = findMenuGroupById(menuRequest);
-
+        final MenuGroup menuGroup = findMenuGroupById(menuRequest.getMenuGroupId());
+        menuValidator.validCreate(menuRequest);
         List<MenuProduct> menuProducts = menuRequest.getMenuProducts()
                 .stream()
-                .map(m -> MenuProduct.of(findProductById(m.getProductId()), m.getQuantity()))
+                .map(m -> MenuProduct.of(m.getProductId(), m.getQuantity()))
```

Comment:
`m` 이라는 축약 네이밍보다 구체적인 네이밍을 활용하거나 `it` 을 사용해보는게 어떨까요?

```suggestion
                .map(it -> MenuProduct.of(it.getProductId(), it.getQuantity()))
```

Reply:
반영했습니다!